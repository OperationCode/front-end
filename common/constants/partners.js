import { s3 } from 'common/constants/urls';

const partners = [
  {
    name: 'APEX Systems',
    logoSource: `${s3}partnerLogos/apex_systems.png`,
    url: 'https://apexsystems.com',
  },
  {
    name: 'GitHub',
    logoSource: `${s3}partnerLogos/github.png`,
    url: 'https://github.com',
  },
  {
    name: '1Password',
    logoSource: `${s3}partnerLogos/1password.png`,
    url: 'https://1password.com/',
  },
  {
    name: 'US Bank',
    logoSource: `${s3}partnerLogos/usbank.png`,
    url: 'https://usbank.com/index.html',
  },
  {
    name: 'Threat Stack',
    logoSource: `${s3}partnerLogos/threat_stack.png`,
    url: 'https://threatstack.com',
  },
  {
    name: "O'Reilly Media",
    logoSource: `${s3}partnerLogos/oreilly.png`,
    url: 'https://oreilly.com',
  },
  {
    name: 'HackerRank',
    logoSource: `${s3}partnerLogos/hackerrank.png`,
    url: 'https://hackerrank.com',
  },
  {
    name: 'Zapier',
    logoSource: `${s3}partnerLogos/zapier.png`,
    url: 'https://zapier.com',
  },
];

export default partners;
