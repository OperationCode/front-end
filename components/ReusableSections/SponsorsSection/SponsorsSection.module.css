.justifyAlign {
  text-align: justify;
}

.verticalSpacing {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 1rem auto;
}

.columnsContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

/* the individual columns */
.columnsContainer > * {
  margin: 1rem;
}
