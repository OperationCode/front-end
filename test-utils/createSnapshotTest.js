/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
