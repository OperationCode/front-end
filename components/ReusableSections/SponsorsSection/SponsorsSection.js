import React from 'react';
import Container from 'components/Container/Container';
import Heading from 'components/Heading/Heading';
import PartnerLogoLink from 'components/PartnerLogoLink/PartnerLogoLink';
import partners, { PARTNER_TYPES } from 'common/constants/partners';
import styles from './SponsorsSection.css';

const isPaidSponsor = partner => partner.type === PARTNER_TYPES.PAID;

const SponsorsSection = () => (
  <Container theme="gray">
    <Heading text="Sponsors" hasTitleUnderline headingLevel={3} />

    <Heading text="Paid Sponsors" headingLevel={4} />
    <p>
      These sponsors have donated cash to help keep the mission going here at Operation Code. They
      may or may not have also donated to our cause via others means.
    </p>
    <div className={styles.columnsContainer}>
      {partners
        .filter(x => isPaidSponsor(x))
        .map(partner => (
          <PartnerLogoLink key={partner.name} {...partner} size="large" />
        ))}
    </div>

    <Heading text="In-Kind Sponsors" headingLevel={4} />
    <p>
      The following organizations have helped Operation Code through services, products,
      advertisements, scholarships, or sponsorships. We thank them for their contributions.
    </p>
    <div className={styles.columnsContainer}>
      {partners
        .filter(x => !isPaidSponsor(x))
        .map(partner => (
          <PartnerLogoLink key={partner.name} {...partner} />
        ))}
    </div>
  </Container>
);

export default SponsorsSection;
