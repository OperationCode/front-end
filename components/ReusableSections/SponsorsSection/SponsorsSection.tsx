import { Container } from 'components/Container/Container';
import { Heading } from 'components/Heading/Heading';
import { PartnerLogoLink } from 'components/PartnerLogoLink/PartnerLogoLink';
import { PARTNER_TYPES, sortedPartners } from 'common/constants/partners';

interface Partner {
  name: string;
  type: string;
}

const isPaidSponsor = (partner: Partner): boolean => partner.type === PARTNER_TYPES.PAID;

export const SponsorsSection = () => (
  <Container theme="gray" data-testid="Sponsors Section">
    <Heading text="Sponsors" hasTitleUnderline headingLevel={3} />

    <Heading text="Corporate Partners" headingLevel={4} />
    <p>
      These sponsors have donated cash to help keep the mission going here at Operation Code. They
      may or may not have also donated to our cause via others means.
    </p>
    <div className="flex justify-center items-center flex-wrap [&>*]:m-4">
      {sortedPartners
        .filter(x => isPaidSponsor(x))
        .map(partner => (
          <PartnerLogoLink key={partner.name} {...partner} size="large" />
        ))}
    </div>

    <Heading text="Corporate Supporters" headingLevel={4} />
    <p>
      The following organizations have helped Operation Code through services, products,
      advertisements, scholarships, or sponsorships. We thank them for their contributions.
    </p>
    <div className="flex justify-center items-center flex-wrap [&>*]:m-4">
      {sortedPartners
        .filter(x => !isPaidSponsor(x))
        .map(partner => (
          <PartnerLogoLink key={partner.name} {...partner} />
        ))}
    </div>
  </Container>
);
