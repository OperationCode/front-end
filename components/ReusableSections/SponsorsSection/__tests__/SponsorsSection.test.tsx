import { render } from '@testing-library/react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import partners from 'common/constants/partners';
import SponsorsSection from '../SponsorsSection';

describe('SponsorsSection', () => {
  it('should render', () => {
    const { queryByTestId } = render(<SponsorsSection />);
    expect(queryByTestId('Sponsors Section')).not.toBeNull();

    createSnapshotTest(<SponsorsSection />);
  });

  it('should render a secure link and image for each partner', () => {
    const component = render(<SponsorsSection />);

    partners.forEach(partner => {
      const image = component.queryByAltText(`${partner.name} logo`)!;
      const link = image.closest('a') as HTMLAnchorElement;
      expect(image).toBeInTheDocument();
      expect(link.href.startsWith('https://')).toStrictEqual(true);
    });
  });
});
