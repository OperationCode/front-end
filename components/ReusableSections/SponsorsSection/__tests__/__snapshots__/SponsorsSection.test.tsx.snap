// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SponsorsSection > should render 1`] = `
<div
  className="bg-center bg-no-repeat bg-cover flex items-center justify-between w-full fill-current min-h-[250px] bg-theme-gray-800 text-secondary"
  data-testid="Sponsors Section"
>
  <div
    className="flex flex-col items-center justify-center mx-auto w-full my-[3.5rem] max-w-[1400px] sm:w-[85%]"
  >
    <div
      className="flex justify-center"
    >
      <h3
        className="group flex uppercase text-center my-4 mx-0 border-solid border-b-4 border-b-primary"
      >
        <div
          className="relative"
          data-testid="Heading Content sponsors-link"
        >
          <a
            className="hidden absolute top-2 -left-8 transition-all duration-200 ease-linear sm:inline sm:opacity-0 sm:group-hover:opacity-100"
            data-testid="Hash Link"
            href="#sponsors-link"
            id="sponsors-link"
          >
            <span
              className="sr-only"
              data-testid="SCREEN_READER_ONLY"
            >
              Scroll Link for 
              Sponsors
            </span>
            <svg
              className="w-4 h-8 leading-9 mx-2 -my-1 outline-none"
              dangerouslySetInnerHTML={
                {
                  "__html": "<symbol xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" id="d456e9ce"><title>Link Icon</title><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"/></symbol><use href='#d456e9ce'/>",
                }
              }
              viewBox="0 0 512 512"
            />
          </a>
          Sponsors
        </div>
      </h3>
    </div>
    <div
      className="flex justify-center"
    >
      <h4
        className="group flex uppercase text-center my-4 mx-0"
      >
        <div
          className="relative"
          data-testid="Heading Content corporate-partners-link"
        >
          <a
            className="hidden absolute top-2 -left-8 transition-all duration-200 ease-linear sm:inline sm:opacity-0 sm:group-hover:opacity-100"
            data-testid="Hash Link"
            href="#corporate-partners-link"
            id="corporate-partners-link"
          >
            <span
              className="sr-only"
              data-testid="SCREEN_READER_ONLY"
            >
              Scroll Link for 
              Corporate Partners
            </span>
            <svg
              className="w-4 h-8 leading-9 mx-2 -my-1 outline-none"
              dangerouslySetInnerHTML={
                {
                  "__html": "<symbol xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" id="d456e9ce"><title>Link Icon</title><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"/></symbol><use href='#d456e9ce'/>",
                }
              }
              viewBox="0 0 512 512"
            />
          </a>
          Corporate Partners
        </div>
      </h4>
    </div>
    <p
      className="mb-5"
    >
      These sponsors have donated cash to help keep the mission going here at Operation Code. They may or may not have also donated to our cause via others means.
    </p>
    <div
      className="flex justify-center items-center flex-wrap gap-10"
    >
      <a
        className="inline-flex items-start"
        href="https://etsy.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[150px] w-[150px]"
          >
            <img
              alt="Etsy logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://www.juniper.net/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[150px] w-[150px]"
          >
            <img
              alt="Juniper Networks logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://threatstack.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[150px] w-[150px]"
          >
            <img
              alt="Threat Stack logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/threat_stack.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://www.usbank.com/index.html?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[150px] w-[150px]"
          >
            <img
              alt="US Bank logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/usbank.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://www.umpquabank.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[150px] w-[150px]"
          >
            <img
              alt="Umpqua Bank logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
    </div>
    <div
      className="flex justify-center"
    >
      <h4
        className="group flex uppercase text-center my-4 mx-0"
      >
        <div
          className="relative"
          data-testid="Heading Content corporate-supporters-link"
        >
          <a
            className="hidden absolute top-2 -left-8 transition-all duration-200 ease-linear sm:inline sm:opacity-0 sm:group-hover:opacity-100"
            data-testid="Hash Link"
            href="#corporate-supporters-link"
            id="corporate-supporters-link"
          >
            <span
              className="sr-only"
              data-testid="SCREEN_READER_ONLY"
            >
              Scroll Link for 
              Corporate Supporters
            </span>
            <svg
              className="w-4 h-8 leading-9 mx-2 -my-1 outline-none"
              dangerouslySetInnerHTML={
                {
                  "__html": "<symbol xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" id="d456e9ce"><title>Link Icon</title><path d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"/></symbol><use href='#d456e9ce'/>",
                }
              }
              viewBox="0 0 512 512"
            />
          </a>
          Corporate Supporters
        </div>
      </h4>
    </div>
    <p
      className="mb-5"
    >
      The following organizations have helped Operation Code through services, products, advertisements, scholarships, or sponsorships. We thank them for their contributions.
    </p>
    <div
      className="flex justify-center items-center flex-wrap gap-10"
    >
      <a
        className="inline-flex items-start"
        href="https://1password.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="1Password logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/1password.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://apexsystems.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="APEX Systems logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/apex_systems.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://airbnb.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Airbnb logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://chromatic.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Chromatic logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/chromatic.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://www.codeplatoon.org/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Code Platoon logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://coursera.org/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Coursera logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/coursera.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://elyoninternational.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Elyon International logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/elyon.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://github.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="GitHub logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/github.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://google.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Google logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://hackerrank.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="HackerRank logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/hackerrank.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://logrocket.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="LogRocket logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/logrocket.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://oreilly.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="O'Reilly Media logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/oreilly.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://securecodewarrior.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Secure Code Warrior logo"
              src="https://user-images.githubusercontent.com/51661129/92973323-3d540b80-f484-11ea-85dd-d202fc751cd9.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://teamtreehouse.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Team Treehouse logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/treehouse.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://uniteus.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Unite Us logo"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://vercel.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Vercel logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/vercel.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
      <a
        className="inline-flex items-start"
        href="https://zapier.com/?utm_source=operationcode"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          <div
            className="relative h-[125px] w-[125px]"
          >
            <img
              alt="Zapier logo"
              src="https://operation-code-assets.s3.us-east-2.amazonaws.com/partnerLogos/zapier.png"
            />
          </div>
          <span
            className="sr-only"
            data-testid="SCREEN_READER_ONLY"
          >
            Opens in new window
          </span>
        </span>
      </a>
    </div>
  </div>
</div>
`;
